### PR TITLE
ssh: avoid crash upon exit-signal

### PR DESCRIPTION
**Pull request by u3s**
- avoid code crash when exit signal is received for closed channel
- closing #8929

### CODE REVIEW

File: lib/ssh/src/ssh_connection.erl [hunk 1]
```diff
@@ -739,21 +739,24 @@ handle_msg(#ssh_msg_channel_request{recipient_channel = ChannelId,
...
+        #channel{remote_id = RemoteId} = Channel ->
+            {Reply, Connection} =  reply_msg(Channel, Connection0,
+                                             {exit_signal, ChannelId,
+                                              binary_to_list(SigName),
+                                              binary_to_list(Err),
+                                              binary_to_list(Lang)}),
+            CloseMsg = channel_close_msg(RemoteId),
+            {[{connection_reply, CloseMsg}|Reply], Connection};
+        _ ->
+            {[], Connection0}
```
Comment (IngelaAndin) [inline lib/ssh/src/ssh_connection.erl:758]: Maybe add a comment %% Connection already closed by peer

Comment (u3s) [reply lib/ssh/src/ssh_connection.erl:758]: you mean channel I guess. yes, good idea. I will.

Comment (IngelaAndin) [reply lib/ssh/src/ssh_connection.erl:758]: Yes I think you are correct about that it should be channel :)  I think I put to much emphasis on the {connection_reply, CloseMsg} part when I read it.  Maybe  
```
ChannelClose  =   channel_close_msg(RemoteId),
{[{connection_reply, ChannelClose}|Reply], Connection};
```
would help shift the attention to what is important? Although it is clear if you think a little longer about it!

Comment (u3s) [reply lib/ssh/src/ssh_connection.erl:759]: yes. applied.

**Pull request merged**